docs/improving-gci-experience.md

```diff
@@ -0,0 +1,377 @@
+require you to claim a task before you start working on it. It's recommended
+since students can't see the number of instances available and it may happen
+that someone else claims the last instance before you can claim it. Your work
+won't then be applicable for review.
+
+The workflow for completing GCI tasks:
+
+* Student claims the task.
+
+* Student works on the tasks and makes sure it fulfills all the requirements.
+
+* Before submitting the task, they check that the task passes all the tests
+(including Travis tests and manual testing of the feature).
+
+* Student submits the task in the GCI app and waits for a review - they
+should focus on other endeavors while waiting.
+
+* The task is going through the review process, with refactoring and
+improvements required by mentors.
+
+* The task is accepted when a mentor assesses it as fulfilling the task
+requirements - this doesn't necessarily mean the pull request will be merged
+at the same time.
+
+## How does the review process work?
+
+It's good to have more than one mentor look at your contribution, since they
+can focus on different aspects of it. The main aspects we focus on with the
+GCI tasks:
+
+* General task requirements - the files are named properly and created in the
+right folders, the commit messages are following both the specific task
+description and [Zulip commit discipline
+doc](http://zulip.readthedocs.io/en/latest/version-control.html#commit-discipline)
+etc.
+
+* Task functionality - the submission implements the required functionality
+appropriately.
+
+* Code quality and style (as in [Zulip code style
+  doc](http://zulip.readthedocs.io/en/latest/code-style.html)).
+
+* Passing the test suite.
+
+* Passing manual tests.
+
+* Files are properly rendering (especially Markdown files).
+
+* User docs adhere to the
+[style guide](http://zulip.readthedocs.io/en/latest/README.html#style-guide).
+
+We also apply rules described in the [Zulip
+review doc](http://zulip.readthedocs.io/en/latest/code-reviewing.html).
+
+## Why is it taking so long for the mentors to check my task?
+
+With reviewing the GCI tasks we prioritize fairness and transparency. We are
+open to helping students whenever they require assistance, but we also keep
+in mind that both GCI and being a great open-source contributor in general
+is about individual contributions and work. It wouldn't be fair to the
+students who are submitting tasks to get blocked because others might be
+more vocal on chat.
+
+We're trying to keep response times to minimum and one of the more objective
+resources we have is the queue in the GCI app (example screenshot below).
+If you've been working on your task and asking for help on chat, you need to
+submit the task and it will be queued in the app. It's not a perfect solution,
+but keeping with the queue assures we are more objective than by simply
+attending only to students active on Zulip chat.
```

Comment:
I think you sorta say this, but it's probably worth mentioning that we try to prioritize submitted tasks over chat if we only have time for one?